pkg/volume/gcepd/attacher.go

```diff
@@ -141,6 +141,45 @@ func (attacher *gcePersistentDiskAttacher) VolumesAreAttached(specs []*volume.Sp
 	return volumesAttachedCheck, nil
 }
 
+func (attacher *gcePersistentDiskAttacher) BulkVerifyVolumes(volumesByNode map[types.NodeName][]*volume.Spec) (map[types.NodeName]map[*volume.Spec]bool, error) {
+	volumesAttachedCheck := make(map[types.NodeName]map[*volume.Spec]bool)
+	deviceNamesByNode := make(map[types.NodeName][]string)
+
+	for nodeName, volumeSpecs := range volumesByNode {
+		deviceNames := []string{}
+		for _, spec := range volumeSpecs {
+			volumeSource, _, err := getVolumeSource(spec)
+			if err != nil {
+				klog.Errorf("Error getting volume (%q) source : %v", spec.Name(), err)
+				continue
+			}
+			deviceNames = append(deviceNames, volumeSource.PDName)
+		}
+		deviceNamesByNode[nodeName] = deviceNames
+	}
+
+	attachedDisksByNode, err := attacher.gceDisks.BulkDisksAreAttached(deviceNamesByNode)
+	if err != nil {
+		klog.Errorf(
```

Comment:
does the err already get logged at a higher level, if so, we can just incoroporate this contextual info into the error returned and remove the logging here

Reply:
This is again something I was following from the current implementation of VolumesAreAttached. I am sure it gets handled by the operation executor / nested pending operation.

Reply:
I removed it, but what about the one on line 128?